src/ape/managers/project.py

```diff
@@ -4,28 +4,271 @@
 import tempfile
 from importlib import import_module
 from pathlib import Path
-from typing import ClassVar, Collection, Dict, List, Optional, Union
+from typing import ClassVar, Dict, List, Optional, Type, Union
 
-import requests
-from ethpm_types import Checksum, Compiler, ContractType, PackageManifest, Source
-from ethpm_types.manifest import PackageName
+import yaml
+from ethpm_types import Compiler, ContractType, PackageManifest
 from ethpm_types.utils import compute_checksum
 
+from ape.api.projects import DependencyAPI, ProjectAPI
 from ape.contracts import ContractContainer
 from ape.exceptions import ProjectError
 from ape.logging import logger
-from ape.managers.networks import NetworkManager
-from ape.utils import (
-    cached_property,
-    get_all_files_in_directory,
-    get_relative_path,
-    github_client,
-    injected_before_use,
-)
+from ape.plugins import PluginManager
+from ape.utils import cached_property, get_relative_path, github_client, injected_before_use
 
 from .compilers import CompilerManager
-from .config import ConfigManager
+from .config import CONFIG_FILE_NAME, ConfigManager
 from .converters import ConversionManager
+from .networks import NetworkManager
+
+
+class ApeProject(ProjectAPI):
+    """
+    The default implementation of the :class:`~ape.api.projects.ProjectAPI`.
+    By default, the `:class:`~ape.managers.project.ProjectManager` uses an
+    ``ApeProject`` at the current-working directory.
+    """
+
+    @property
+    def is_valid(self) -> bool:
+        if (self.path / CONFIG_FILE_NAME).exists():
+            return True
+
+        logger.warning(
+            f"'{self.path.name}' is not an 'ApeProject', but attempting to process as one."
+        )
+
+        # NOTE: We always return True as a last-chance attempt because it often
+        # works anyway and prevents unnecessary plugin requirements.
+        return True
+
+    @property
+    def sources(self) -> List[Path]:
+        """
+        All the source files in the project.
+        Excludes files with extensions that don't have a registered compiler.
+
+        Returns:
+            List[pathlib.Path]: A list of a source file paths in the project.
+        """
+
+        files: List[Path] = []
+        if not self.contracts_folder.exists():
+            return files
+
+        for extension in self.compilers.registered_compilers:
+            files.extend(self.contracts_folder.rglob(f"*{extension}"))
+
+        return files
+
+    def create_manifest(
+        self, file_paths: Optional[List[Path]] = None, use_cache: bool = True
+    ) -> PackageManifest:
```

Comment:
Does this need a doc string?

Reply:
By not including one, it will automatically used the doc str from the abstractmethod in `ProjectAPI`, which seems desirable. It is not intuitive that is happening but it cuts down on repetitive documentation

Reply:
Missed that this was at the abstract level, thanks!